components/hab/src/main.rs

```diff
@@ -841,6 +843,51 @@ fn sub_pkg_sign(ui: &mut UI, m: &ArgMatches<'_>) -> Result<()> {
     command::pkg::sign::start(ui, &pair, &src, &dst)
 }
 
+fn sub_pkg_bulkupload(ui: &mut UI, m: &ArgMatches<'_>) -> Result<()> {
+    let upload_dir = bulkupload_dir_from_matches(m);
+    // upload_dir is required via clap and exists
+    let artifact_path = upload_dir.as_ref().unwrap().join("artifacts");
+    let key_path = upload_dir.as_ref().unwrap().join("keys");
+    let url = bldr_url_from_matches(&m)?;
```

Comment:
`m` is already a reference to `ArgMatches`, so there's no need to take an additional reference:
```suggestion
    let url = bldr_url_from_matches(m)?;
```
Similarly with `channel_from_matches` and `auth_token_param_or_env`.

Reply:
Addressed in https://github.com/habitat-sh/habitat/pull/6958/commits/bc5b9642d03ce480fd4573ae05eaef724b3c12a7